_summaries/ba2016layer_1.md

```diff
@@ -1,19 +1,66 @@
+
+* Additionally, through analyzing the manifold of the parameter space, the paper also shows that normalization has an implicit effect of reducing the learning rate and making the learning more stable.
+
+
+## How is it realized (technically)?
+
+Ioffe and Szegedy (2015) had proposed the idea of eliminating the effect of covariate shift by normalizing the summed inputs to a neuron before applying the non-linear activation on top of that. They wanted to normalize the inputs to each neuron independently using mean and variance (of the inputs) computed over the distribution of the data. So, 
+
+$$ a_i^l = {w_i^l}^T h^l \qquad h_i^{l+1} = f(a_i^l + b_i^l)$$
+
+changed to
+
+$$ \overline{a}_i^l = \frac{g_i^l}{\sigma_i^l}(a_i^l-\mu_i^l) \qquad \mu_i^l = \underset{\mathrm{x}\sim P(\mathrm{x})}{\mathbb{E}}[a_i^l] \qquad \sigma_i^l = \sqrt{\underset{\mathrm{x}\sim P(\mathrm{x})}{\mathbb{E}} \left [\left ( a_i^l-\mu_i^l \right )^2 \right ] }$$
+
+where $a_i^l$ are the summed inputs to neuron $i$ in layer $l$, $w_i^l$ is the incoming weight vector for that neuron, $h^l$ is the output from previous layer, $b_i^l$ is the bias term, $f(\cdot)$ is the activation of the neuron, $\mathrm{x}$ is the input data, $g_i^l$ is the gain parameter and $\overline{a}_i^l$ is the normalized sum of inputs. 
+
+Obviously, it is impractical to compute the above expectations exactly, hence the batch normalization paper computes it empirically over the samples in the current mini-batch. This layer normalization paper modifies that and computes the normalization statistics (mean and variance) over all the neurons in the layer for each training sample:
+
+$$ \mu^l = \frac{1}{H} \sum_{i=1}^{H}a_i^l \qquad \sigma^l = \sqrt{\frac{1}{H}\sum_{i=1}^{H} \left ( a_i^l-\mu_i^l\right )^2}$$
+
+Note that the mean and variance above are defined for a layer and are same for all neurons in that layer.
+
+### Why is layer normalization superior to batch normalization?
+
+* As shown above, batch normalization computes the normalization parameters over a mini-batch of samples which makes it sensitive to the batch size. Therefore, one cannot reduce the batch size arbitrarily. Layer normalization, on the other hand, does not put a constraint on the batch size during training and hence, can be applied even with batch size = 1 as in online learning.
+* For reasons similar to that in previous point, it is not easy to apply batch normalization in RNNs because one would need to compute and store the parameters for **each time-step across sequences** in a batch and that is a problem when the sequence length varies a lot as it commonly does. Layer normalization does not run into this problem because it computes the statistics from the inputs to all nodes in layer at the **current time-step**.
+
```

Comment:
You can also talk a bit about how the normalization reduces learning rate, and stabilizes learning. Maybe add equation 14.

Reply:
Don't you think the summary is longer than average right now? I will include the point, but skip the equation probably because I'll have to also explain it a bit if I add it.

Reply:
Yeah sure no problem, you can skip it.